pages/admin/lessons/[lessonId]/[pageName]/index.tsx

```diff
@@ -0,0 +1,179 @@
+  order: number
+}
+type Modules = Module[]
+
+type ReqParam = string | string[] | undefined
+
+type ContentProps = {
+  pageName: ReqParam
+  modules: Modules
+  lessonId: number
+  refetch: Props['refetch']
+}
+
+const extractLessonOrder = (x: string) => x.split('').slice(2).join()
+
+const Content = ({ pageName, modules, lessonId, refetch }: ContentProps) => {
+  const [selectedIndex, setSelectedIndex] = useState(-1)
+  const onAddItem = () => setSelectedIndex(-1)
+  const onSelect = (item: Omit<Module, 'order'>) => setSelectedIndex(item.id)
+
+  if (pageName !== 'modules') {
+    return (
+      <h1>
+        For now, you can only access <code>/modules</code> page
+      </h1>
+    )
+  }
+
+  return (
+    <div className={styles.container__modulesPanel}>
+      <AdminLessonSideNav
+        title="Modules"
+        items={modules}
+        onAddItem={onAddItem}
+        onSelect={onSelect}
+        selectedIndex={selectedIndex}
+      />
+      <div className={styles.container__modulesPanel__inputs}>
+        <AdminModuleInputs
+          lessonId={lessonId}
+          refetch={refetch}
+          module={modules.find(module => module.id === selectedIndex)}
+        />
+      </div>
+    </div>
+  )
+}
+
+const Lessons = ({ data }: GetAppProps) => {
+  const router = useRouter()
+  const { pageName, lessonId } = router.query
+
+  const { lessons } = data
+  const { data: modulesData, refetch } = useQuery<{ modules: Modules }>(MODULES)
+
+  const lesson = useMemo(() => {
+    if (lessons) {
+      const lessonFromParam = lessons.find(
+        e => toNumber(extractLessonOrder(e.slug)) === toNumber(lessonId)
+      )
+
+      if (lessonFromParam)
+        return {
+          title: lessonFromParam.title,
+          slug: lessonFromParam.slug,
+          id: lessonFromParam.id
+        }
+    }
+
+    return { title: '', slug: '', id: -1 }
+  }, [lessons, lessonId])
+
+  const modules = useMemo(
+    () => sortBy(get(modulesData, 'modules'), 'order'),
+    [modulesData]
+  )
+
+  const filteredModules = useMemo(
+    () => modules.filter(module => module.lesson.id === lesson.id),
+    [lesson.id, modules]
+  )
+
+  const LessonNav = ({
+    tab
+  }: {
+    tab: { tabName: string; urlPageName: string }
+  }) => {
+    const isSelected = tab.urlPageName === pageName
+    const className =
+      navStyles[
+        isSelected
+          ? 'lessons_tabsNav__nav__item'
+          : 'lessons_tabsNav__nav__item--inactive'
+      ]
+
+    return (
+      <Link href={`${MAIN_PATH}/${lessonId}/${tab.urlPageName}`}>
+        <a className={`${className} nav-pills`}>{toUpper(tab.tabName)}</a>
+      </Link>
+    )
+  }
+
+  return (
+    <AdminLayout data={data}>
+      <main className={styles.container}>
+        <header>
+          <h1>Lesson Info</h1>
+          <Breadcrumbs
+            lesson={lesson}
+            setLesson={lesson => {
+              router.push(
+                `${MAIN_PATH}/${extractLessonOrder(lesson.slug)}/${pageName}`
```

Comment:
if we are now using lesson.slug, should the file name be `/pages/admin/lessons/[lessonSlug]/[page...`